src/http_server.cc

```diff
@@ -3126,17 +3120,23 @@ HTTPAPIServer::InferRequestClass::InferResponseComplete(
   }
 #endif  // TRITON_ENABLE_TRACING
 
+  LOG_TRITONSERVER_ERROR(
+      TRITONSERVER_InferenceResponseDelete(response),
+      "deleting inference response");
+
+  // Defer sending the response until FINAL flag is seen or
+  // there is error
+  if ((err == nullptr) && (flags & TRITONSERVER_RESPONSE_COMPLETE_FINAL) == 0) {
```

Comment:
I made the change to not send error until FINAL flag is seen, otherwise there can be issue on the backend trying to call the response send function again after the `userp` has been released. Currently the only assumption is that the backend will not call the function again once final flag is sent.